### DISCUSSION TITLE
Make task_instance priority_weight dag_run/execution_date dependent

### DISCUSSION THREAD

**Discussion by aaronmielke** [apache/airflow#24835] [General]
### Description

We want to weight a task's priority based on execution date. In general it would be nice to have the option to make the priority weight of a task instance dynamic based on the dag_run (from which execution_date and possibly other parameters can be derived)

### Use case/motivation

We have a pool to run a set of tasks. Sometime we rerun a lot of old dag instances. This queues a LOT of task instances. Those dag rerun task instances end up delaying "today's" dag_instance tasks. If we weight today's dag_instance tasks with a higher priority weight this wouldn't happen.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
This would be a major change in the architectuer of scheduler and executor. I think it is not enough to make it a feature. If someone would like to implement such prioritisation that would require major re-architecting of how schedulers and executors work. That requires AIP https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals and extended discussion on devlist. converting this into discussion.

**Comment by ht-jo** [comment]
Hello. @potiuk I'm using dynamic task mapping, I tried controlling its priorities of a task map. As you know, the priority property is one of the partial method properties, so It can't be controlled. I tried intercepting and manipulating values, but I'm not 100% python programmer, I couldn't find out a way to change priority values.

I think there should be a way to be able to change priorities of a task map. Also I respect how to work Airflow scheduler and executors by its philosophy, It'll be more great that if developers who needs to control can control priorities of a task map.

**Reply by potiuk** [reply]
Well. If someone proposes and implements it of - course. But it needs somoene to do it. This is open-source and unless there is somoene stepping up and doing it, it won't happen. There is no-one who can tell "let's do it". If you really want it - I propose you to write at the devlist and ask if there is somoene willing to do it and offer whatever help you can.

Just - to set expectations, features here do not as as "backlog" that gets implemented. In Open Source, someone (and that can be anyone - you, me, anyone) can say "I want to do it" propose AIP, get to consensus, pass voting and implement it.

I am not trying to "downplay it" - it might be interesting capability, but unless someone (even listening to the conversation) steps up, this is at most a discussion. You do not need to convince me :). I am not against the proposal at all (if I were I would have told why and argue). If you do not feel enough on doing it, and you want to do it, you need to find someone who will. This is really how everything works here :). Raising an idea is a discussion. Designing and implementing is when it materializes.

**Reply by ht-jo** [reply]
@potiuk  I learned how to operate open-source ecosystem from your comment. I just wrote my complaints.. I'll escalate my idea. Thanks for your feedback.
